src/components/Form.js

```diff
@@ -1,14 +1,55 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+// import { nanoid } from '@reduxjs/toolkit';
+import { v4 as uuidv4 } from 'uuid';
 import styles from '../styles/Form.module.css';
+import { addBook, bookPost } from '../features/book/Bookslice';
 
 function Form() {
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+
+  const onTitleChanged = (e) => setTitle(e.target.value);
+  const onAuthorChanged = (e) => setAuthor(e.target.value);
+
+  const dispatch = useDispatch();
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (!title.trim() || !author.trim()) return;
+    const bookData = {
+      item_id: uuidv4(),
+      title,
+      author,
+
+    };
+    dispatch(addBook(bookData));
+    dispatch(bookPost(bookData));
+    setTitle('');
+    setAuthor('');
+  };
```

Comment:
- [ ] When I add a book, it gets added to the store and rendered on the screen, but when the browser is reloaded with the expectation of the data being persisted to the API, the data is not saved to the API. Kindly check that out and be sure any book added is saved to the API, to the store, and is rendered as expected 👍🏽

Reply:
resolved! ✔️